app/src/androidTest/java/org/simple/clinic/overdue/AppointmentRepositoryAndroidTest.kt

```diff
@@ -889,64 +1047,75 @@ class AppointmentRepositoryAndroidTest {
       )
     }
 
-    fun createPatient(fullName: String): PatientProfile {
-      return testData.patientProfile(generatePhoneNumber = true)
-          .let { patientProfile ->
-            patientProfile.copy(patient = patientProfile.patient.copy(fullName = fullName))
-          }
-    }
-
-    fun scheduleAppointment(patientProfile: PatientProfile): Single<Appointment> {
+    fun scheduleAppointment(
+        appointmentUuid: UUID,
+        patientProfile: PatientProfile
+    ): Single<Appointment> {
       return appointmentRepository.schedule(
           patientUuid = patientProfile.patient.uuid,
-          appointmentDate = LocalDate.now(clock).minus(2, ChronoUnit.DAYS),
+          appointmentUuid = appointmentUuid,
+          appointmentDate = LocalDate.parse("2017-12-30"),
           appointmentType = Manual,
-          currentFacility = facility)
+          currentFacility = facility
+      )
     }
 
-    val patient1 = createPatient("Patient with one latest updated retro-active BP")
-    val patient2 = createPatient("Patient with no retro-active BP")
+    // given
+    val now = Instant.now(clock)
+
+    val firstPatient = testData.patientProfile(
+        patientUuid = UUID.fromString("e1943cfb-faf0-42c4-b5b6-14b5153295b2"),
+        generatePhoneNumber = true
+    )
+    val secondPatient = testData.patientProfile(
+        patientUuid = UUID.fromString("08c7acbf-61f1-439e-93a8-43ba4e990428"),
+        generatePhoneNumber = true
+    )
 
-    patientRepository.save(listOf(patient1, patient2)).blockingAwait()
+    patientRepository.save(listOf(firstPatient, secondPatient)).blockingAwait()
 
-    val bpOneForPatient1 = createBloodPressure(
-        patientProfile = patient1,
+    val earlierRecordedBpForFirstPatient = createBloodPressure(
+        patientProfile = firstPatient,
+        recordedAt = now.minusSeconds(1),
+        updatedAt = now.plusSeconds(1)
+    )
+    val laterRecordedBpForFirstPatient = createBloodPressure(
+        patientProfile = firstPatient,
         recordedAt = now,
         updatedAt = now
     )
-    val bpTwoForPatient1 = createBloodPressure(
-        patientProfile = patient1,
-        recordedAt = now.minus(10, ChronoUnit.DAYS),
-        updatedAt = now.plusSeconds(10)
-    )
 
-    val bpOneForPatient2 = createBloodPressure(
-        patientProfile = patient2,
+    val earlierRecordedBpForSecondPatient = createBloodPressure(
+        patientProfile = secondPatient,
         recordedAt = now,
-        updatedAt = now)
-
-    val bpTwoForPatient2 = createBloodPressure(
-        patientProfile = patient2,
-        recordedAt = now.plus(10, ChronoUnit.MINUTES),
+        updatedAt = now.plusSeconds(1)
+    )
+    val laterRecordedBpForSecondPatient = createBloodPressure(
+        patientProfile = secondPatient,
+        recordedAt = now.plusSeconds(1),
```

Comment:
All these `earlier` and `later` records can be created using `testClock.advanceBy()` instead of `plusSeconds()`.
We can create all `earlier` ones and then advance the clock and then create `later` ones.

Reply:
What benefit does doing this via the clock give us? I feel that this way of representing this is more readable. 

Alternately, I can explicitly parse an `Instant` and see if that makes this test more explicit about what is actually happening behind the scenes.

Reply:
Done.

Reply:
It is closer to the real implementation. That's the advantage of having a `testClock` in the first place.
Plus you don't have to set explicit values for `recordedAt`. You can just use `Instant.now`.

Reply:
In this specific test, we should not care about whether the BPs are recorded using the clock or not. What we care about when fetching overdue appointments is that the latest recorded BP is picked. 

Which means that it is better for us to remove the variability of the clock from this test and explicitly set the time of the BPs ourselves. What do you think?

Reply:
Ok. Makes sense.